woocommerce/woocommerce-ios
# Order list: implemented the screens for showing the date range filters

Part of https://github.com/woocommerce/woocommerce-ios/issues/5243

## Description
In this PR, I implemented the login for showing the filters and for passing data between the flow. I also implemented the login for managing the custom date range filter. The PR is big because of the changes in the various `xib` files.

## Changes
- Added `ordersDateRange` between the options of `FilterListViewController`. This will allow us to present the view controller for filtering orders by dates.
- Implemented `OrderDatesFilterViewController` which allow us to select different options of dates (any, 2 days ago, etc…)
- Implemented `DateRangeFilterViewController`, which allow us to filter the orders by a range of dates, following the new design proposed by Adam.
- Implemented the new struct `OrderDateRangeFilter` (updated the old `OrderDateRangeFilterEnum`) which contains the various type (enum) of Date Range plus the start and end date in case the user will choose a custom range.
- Updated the constraints of `DatePickerTableViewCell` lateral constraints were incorrect.

## Testing
*Important*: Currently the filters are not filtering the orders. This will be part of another PR.
1. Launch the app it in debug (because of the feature flag).
2. Open the order list.
3. You should be able to tap the `Filter` button.
4. You should be able to view the `Filters` screen with two options: Order Status and Date Range.
5. Under Date Range, you should be able to select just one filter.
6. Tap the custom date range, you should be able to select just the start date, just the end date or both.
7. Go back, you should be able to see the dates selected in the Date Range screen.
8. Go back, you should be able to see the dates selected in the Filters screen.
9. You should be able also to change the date filters.

## Video
https://user-images.githubusercontent.com/495617/140803442-097bc519-7b19-44e6-bc9f-930b9d9174e8.mp4


Update release notes:
